vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts

```diff
@@ -0,0 +1,303 @@
+    If we were to install another .NET, we would break your custom .NET installation, so the installation request has been refused.
+    If you would like to proceed with installing .NET automatically for VS Code, you must remove this custom installation.
+    Your custom install is located at: `;
+    public baseUnsupportedDistroErrorMessage = 'We are unable to detect the distro or version of your machine';
+
+    constructor(acquisitionContext : IAcquisitionWorkerContext, executor : ICommandExecutor | null = null, distroProvider : IDistroDotnetSDKProvider | null = null)
+    {
+        this.commandRunner = executor ?? new CommandExecutor(acquisitionContext.eventStream);
+        this.acquisitionContext = acquisitionContext;
+        this.versionResolver = new VersionResolver(acquisitionContext.extensionState, acquisitionContext.eventStream,
+            acquisitionContext.timeoutValue, acquisitionContext.proxyUrl);
+        if(distroProvider)
+        {
+            this.distroSDKProvider = distroProvider;
+        }
+    }
+
+    /**
+     * @remarks relies on /etc/os-release currently. public for testing purposes.
+     * @returns The linux distro and version thats running this app. Should only ever be ran on linux.
+     */
+    public async getRunningDistro() : Promise<DistroVersionPair>
+    {
+        if(this.distro)
+        {
+            return this.distro;
+        }
+
+        const commandResult = (await this.commandRunner.execute('cat /etc/os-release'))[0];
+        const distroNameKey = 'NAME';
+        const distroVersionKey = 'VERSION_ID';
+
+        try
+        {
+            const stdOut = commandResult.toString().split('\n');
+            // We need to remove the quotes from the KEY="VALUE"\n pairs returned by the command stdout, and then turn it into a dictionary. We can't use replaceAll for older browsers.
+            // Replace only replaces one quote, so we remove the 2nd one later.
+            const stdOutWithQuotesRemoved = stdOut.map( x => x.replace('"', ''));
+            const stdOutWithSeparatedKeyValues = stdOutWithQuotesRemoved.map( x => x.split('='));
+            const keyValueMap =  Object.fromEntries(stdOutWithSeparatedKeyValues.map(x => [x[0], x[1]]));
+
+            // Remove the 2nd quotes.
+            const distroName : string = keyValueMap[distroNameKey]?.replace('"', '') ?? '';
+            const distroVersion : string = keyValueMap[distroVersionKey]?.replace('"', '') ?? '';
+
+            if(distroName === '' || distroVersion === '')
+            {
+                const error = new DotnetAcquisitionDistroUnknownError(new Error(this.baseUnsupportedDistroErrorMessage));
+                this.acquisitionContext.eventStream.post(error);
+                throw error.error;
+            }
+
+            const pair : DistroVersionPair = { distro : distroName, version : distroVersion };
+            return pair;
+        }
```

Comment:
if you don't want to do this parsing you can check out the `lsb_release` utility. if you run `lsb_release --release --id --short 2>/dev/null` you get the distro name (e.g. Ubuntu) and the distro version (e.g. 23.10) in one call, with no odd key/value parsing required:

```terminal
$ lsb_release --release --short --id 2>/dev/null
Ubuntu
23.10
```

Reply:
That's a really cool idea, perhaps we could do ls /etc/*release and use that directory as well. Some distros and older versions of even ubuntu or fedora, such as fedora 14, do not have lsb_release, so I am not sure I would want to rely on that being there.